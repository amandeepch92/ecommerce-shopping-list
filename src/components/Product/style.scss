@import 'assets/scss/variables.scss';
.product-details{
    border: 1px solid $lightGray;
    list-style: none;
    display: flex;
    flex: 1 1 300px;
    flex-wrap: wrap;
    padding: 30px 10px;
    padding-top: 20px;
    flex-direction: column;
    justify-content: space-between;
    flex-grow: 0;
  
    .sale-tag {
      background-color: $red;
      color: $white;
      padding: 16px 21px;
      display: inline-block;
    }
  
    .exclusive-tag {
      background-color: $green;
      color: $white;
      padding: 16px 35px;
      display: inline-block;
    }
  
    .product-description {
      display: flex;
      justify-content: space-between;
      margin-top: 30px;
      align-items: baseline;
    }
    .product-name {
      font-size: 17px;
      max-width: 170px;
      font-weight: bold;
    }
    .product-price {
      font-size: 22px;
      font-weight: bold;
    }
}