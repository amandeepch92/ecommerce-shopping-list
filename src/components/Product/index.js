import PropTypes from "prop-types";
import { exclusive, sale } from "locales/en";
import Image from "components/Image";
import "./style.scss";


/**
 *  Product component can be used to display a Product. This component is connected with Styled component
 *  @param {function} item : Item object contains information individual product information
 */
export default function Product(props) {
  const { item, className } = props;
  return (
    <li className={className}>
      <Image src={item.productImage} alt={item.productName} />
      <div>
        {item.isExclusive && <div className="exclusive-tag">{exclusive} </div>}
        {item.isSale && <div className="sale-tag">{sale} </div>}
        <div className="product-description">
          <div className="product-name">{item.productName}</div>
          <div className="product-price">{item.price}</div>
        </div>
      </div>
    </li>
  );
};

Product.propTypes = {
  item: PropTypes.object,
  className: PropTypes.string
};
