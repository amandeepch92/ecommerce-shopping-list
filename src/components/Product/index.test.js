import { shallow } from 'enzyme';
import Product from './index';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
const { configure } = require("enzyme");
configure({ adapter: new Adapter() });

const props = {
    item: {
      index: 0,
      isSale: true,
      isExclusive: false,
      price: "$18.88",
      productImage: "product-1.jpg",
      productName: "Striped shirt",
      size: ["S", "L", "XL"]
    }
  };
  
  describe("Product Component", () => {
    test("should render correctly", () => {
      const product = shallow(<Product {...props} />);
      expect(product).toMatchSnapshot();
    });
  });
  