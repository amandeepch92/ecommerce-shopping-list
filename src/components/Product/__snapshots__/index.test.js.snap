// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component should render correctly 1`] = `
<li>
  <Image
    alt="Striped shirt"
    src="product-1.jpg"
  />
  <div>
    <div
      className="sale-tag"
    >
      Sale
       
    </div>
    <div
      className="product-description"
    >
      <div
        className="product-name"
      >
        Striped shirt
      </div>
      <div
        className="product-price"
      >
        $18.88
      </div>
    </div>
  </div>
</li>
`;
